docker-app/qfieldcloud/core/models.py

```diff
@@ -1415,7 +1421,65 @@ class Meta:
         managed = False
 
 
-class Delta(models.Model):
+class AbstractValidatedProjectOwner(models.Model):
+    """ "
+    Assumes a foreign key field 'project' for a Project
+    """
+
+    class Meta:
+        abstract = True
+
+    @property
+    def account(self):
+        return self.project.owner.useraccount
+
+    @property
+    def is_supported_regarding_online_vector_data(self) -> bool:
+        return (
+            not self.project.has_online_vector_data
+            or self.account.current_subscription.plan.is_external_db_supported
+        )
+
+    @property
+    def is_active(self) -> bool:
+        return self.account.current_subscription.is_active
+
```

Comment:
When we extend this as a `mixin` in `Job` or `Delta`, it means there will be `job.is_active` and `delta.is_active`, which will be a bit misleading. I would propose we remove this one.

Reply:
```suggestion
```